backend/src/controllers/v1/authController.ts

```diff
@@ -267,3 +301,753 @@ export const getNewToken = async (req: Request, res: Response) => {
+        clientSecretTTL: ttl,
+        isClientSecretRevoked: false
+    }).save();
+
+    await EEAuditLogService.createAuditLog(
+        req.authData,
+        {
+            type: EventType.CREATE_IDENTITY_UNIVERSAL_AUTH_CLIENT_SECRET,
+            metadata: {
+                identityId: identityMembershipOrg.identity._id.toString(),
+                clientSecretId: identityUniversalAuthClientSecret._id.toString()
+            }
+        }
+    );
+
+    return res.status(200).send({
+        clientSecret,
+        clientSecretData: packageUniversalAuthClientSecretData(identityUniversalAuthClientSecret)
+    });
+}
+
+export const getUniversalAuthClientSecrets = async (req: Request, res: Response) => {
+    const {
+        params: { identityId }
+    } = await validateRequest(reqValidator.GetUniversalAuthClientSecretsV1, req);
+    
+    const identityMembershipOrg = await IdentityMembershipOrg.findOne({
+        identity: new Types.ObjectId(identityId)
+    }).populate<{
+        identity: IIdentity,
+        customRole: IRole
+    }>("identity customRole");
+
+    if (!identityMembershipOrg) throw ResourceNotFoundError();
+
+    const { permission } = await getAuthDataOrgPermissions({
+        authData: req.authData,
+        organizationId: identityMembershipOrg.organization
+    });
+    ForbiddenError.from(permission).throwUnlessCan(
+        OrgPermissionActions.Read,
+        OrgPermissionSubjects.Identity
+    );
+
+    if (identityMembershipOrg.identity?.authMethod !== IdentityAuthMethod.UNIVERSAL_AUTH) throw BadRequestError({
+        message: "The identity does not have universal auth configured"
+    });
+
+    const rolePermission = await getOrgRolePermissions(
+        identityMembershipOrg?.customRole?.slug ?? identityMembershipOrg.role,
+        identityMembershipOrg.organization.toString()
+    );
+    const hasRequiredPrivileges = isAtLeastAsPrivilegedOrg(permission, rolePermission);
+
+    if (!hasRequiredPrivileges) throw ForbiddenRequestError({
+        message: "Failed to get client secrets for more privileged MI"
+    });
+
+    const clientSecretData = await IdentityUniversalAuthClientSecret
+        .find({
+            identity: identityMembershipOrg.identity,
+            isClientSecretRevoked: false
+        })
+        .sort({ createdAt: -1 })
+        .limit(5);
+
+    await EEAuditLogService.createAuditLog(
+        req.authData,
+        {
+            type: EventType.GET_IDENTITY_UNIVERSAL_AUTH_CLIENT_SECRETS,
+            metadata: {
+                identityId: identityMembershipOrg.identity._id.toString()
+            }
+        }
+    );
+
+    return res.status(200).send({
+        clientSecretData: clientSecretData.map((clientSecretDatum) => packageUniversalAuthClientSecretData(clientSecretDatum))
+    });
+}
+
+export const revokeUniversalAuthClientSecret = async (req: Request, res: Response) => {
+    const {
+        params: { identityId, clientSecretId }
```

Comment:
I'd keep it at the moment; this keeps the pattern RESTful as a sub-resource and also in case `clientSecretId` can be same across different identities.